gms/gossiper.cc

```diff
@@ -1459,10 +1459,8 @@ void gossiper::update_timestamp_for_nodes(const std::map<inet_address, endpoint_
 }
 
 void gossiper::mark_alive(inet_address addr, endpoint_state& local_state) {
-    // if (MessagingService.instance().getVersion(addr) < MessagingService.VERSION_20) {
-    //     real_mark_alive(addr, local_state);
-    //     return;
-    // }
+    // Enter the _background_msg gate so stop() would wait on it
+    auto gh = _background_msg.hold();
```

Comment:
We can put this line right before:

(void)_messaging.send_gossip_echo(...)

Reply:
ok